doc/man/sphinx-build.rst

```diff
@@ -4,14 +4,20 @@ sphinx-build
 Synopsis
 --------
 
-**sphinx-build** [*options*] <*sourcedir*> <*outputdir*> [*filenames* ...]
+| **sphinx-build** [*options*] <*sourcedir*> <*outputdir*> [*filenames* ...]
+| **sphinx-build** -M <*builder*> <*sourcedir*> <*outputdir*> [*options*]
```

Comment:
```suggestion
**sphinx-build** [-M <*builder*>] <*sourcedir*> <*outputdir*> [*filenames* ...] [*options*]
```

Reply:
@AA-Turner unfortunately the last suggestion is misleading. The whole point of this PR is to show that signatures differ. if we have an -M flag then options go AFTER positionals. If there is no -M then they go BEFORE positionals. This is precisely the reason why current docs are confusing and why there is a bunch of questions on stackoverflow addressing "broken" optionals.

Reply:
> If there is no -M then they go BEFORE positionals.

Why is this? Options can go after positionals with no `-M`. See #12795 to add tests.

Reply:
> Why is this? Options can go after positionals with no `-M`.

Ok, fair point, `-b` mode does not care that much about optionals' position because it delegates parsing to `argparse`. But the main point is when we use `-M` we MUST provide optionals after positionals and so far it is never stated in a clear form anywhere. Which is an issue in my opinion, at least for people who think like me and tend to stack optionals together.

An examle to illustrate the issue:
`sphinx-build -M html -v -n sphinx build/html` will lead to `run_make_mode()` getting `['-b', 'html', '-d', '-n\\doctrees', '-v', '-n\\html', 'sphinx', 'build/html']`. This happens because `Make()` expects *src* and *dst* to immediately follow the `-M` flag.
This leads to a lot of frustration on part of anyone who just starts working with the tool. In my case it was `sphinx-build -M html -d build/doctree sphinx build/html` which is even whorse because it would not fail but put doctree into a `build/doctree/html` dir. So each specific build would rebuild doctrees. It took me 3 hours of debugging to figure out the issue. And how many people won't even try to pinpoint the issue?

So my argument is this should be documented. Either the fact that `-M` requires positionals to immediately follow it or to present both signatures as different usecases. To be fair I would actually prefer to:
- **a)** split them into 2 subcommands
- **b)** move `-M` case processing to `argparse` and make `-M|b` mutually excluseive.

The issue though is it would be a breaking change and as much as I like the idea, I don't want pipelines to crash. So I opted for an explicit documentation that would cover both official docs and an `-h` flag so no matter what you choose, you get that info and don't have to waste 3 hours figuring out why the tool does not work as expected.

Now, getting back to your suggestion: It would work, yes. But it still would not make people treat `-M` and `-b` as very different and mutually exclusive flags as they are. So I still prefer 2 signatures because it is explicit and leaves a lot less room for an error. Yes, the `-b` signature might have some adjustments to address the flexibility of optionals' positioning, but IMO it still should be clearly stated that `-M` and `-b` are 2 VERY different things and should be treated accordingly. And this should not come in a form of a note hidden somewhere far down.
I personally still have issues getting direct links to an official sphinx-documentation in my search results. I literally had to open the source code, find a couple key phrases and search by those to find the docs page in question. So I would prefer this info being present very close to any signature we have, be it the docs or an `-h` output.

Reply:
@AA-Turner any feedback on the issue please?